bentoml/_internal/bento/build_config.py

```diff
@@ -430,9 +624,9 @@ def with_defaults(self) -> "FilledBentoBuildConfig":
             {} if self.labels is None else self.labels,
             ["*"] if self.include is None else self.include,
             [] if self.exclude is None else self.exclude,
-            (DockerOptions() if self.docker is None else self.docker).with_defaults(),
-            (PythonOptions() if self.python is None else self.python).with_defaults(),
-            (CondaOptions() if self.conda is None else self.conda).with_defaults(),
+            OptionsFactory.with_defaults(self.docker, DockerOptions),
+            OptionsFactory.with_defaults(self.python, PythonOptions),
+            OptionsFactory.with_defaults(self.conda, CondaOptions),
```

Comment:
Why not just change the defaults for these to empty options?

Reply:
fixed